src/annotations/AnnotationManager.php

```diff
@@ -299,6 +306,22 @@ protected function getAnnotations($class_name, $member_type = self::MEMBER_CLASS
                 }
             }
 
+            if ($this->_traitsSupported && $member_name) {
+                $traitAnnotations = array();
+
+                if (isset($file->traitMethodOverrides[$class_name][$member_name])) {
+                    list($traitName, $originalMemberName) = $file->traitMethodOverrides[$class_name][$member_name];
+                    $traitAnnotations = $this->getAnnotations($traitName, $member_type, $originalMemberName);
+                } else {
+                    foreach ($reflection->getTraitNames() as $traitName)
+                    {
+                        $traitAnnotations += $this->getAnnotations($traitName, $member_type, $member_name);
```

Comment:
1. If multiple traits are used, but `$member_name` is defined in one of them, then we're calling `getAnnotations` on other traits for nothing. This may even produce some notices.
2. If same member (property/method) is defined in more than one trait and no `insteadof` or `as` is used to resolve an conflict, then we should throw an exception here. I suggest using `getTraits` instead of `getTraitNames` method to get both trait names and their `ReflectionClass` class object for easy checking for method presence.